R/pkg/tests/fulltests/test_sparkSQL.R

```diff
@@ -1382,8 +1382,8 @@ test_that("column functions", {
   c20 <- to_timestamp(c) + to_timestamp(c, "yyyy") + to_date(c, "yyyy")
   c21 <- posexplode_outer(c) + explode_outer(c)
   c22 <- not(c)
-  c23 <- trunc(c, "year") + trunc(c, "yyyy") + trunc(c, "yy") +
-    trunc(c, "month") + trunc(c, "mon") + trunc(c, "mm")
+  c23 <- trunc(to_date(c), "year") + trunc(to_date(c), "yyyy") + trunc(to_date(c), "yy") +
+    trunc(to_date(c), "month") + trunc(to_date(c), "mon") + trunc(to_date(c), "mm")
```

Comment:
Ah, just to be clear, I meant https://github.com/apache/spark/pull/18106/files/7fee61b1e084a1ae9966e7ad62b1509085b24151#r123758795The current state looks drop supporting this function with string that can be implicitly casted into date and `to_date` is required which breaks backward compatibility.